### PR TITLE
Allow to select multiple label sets for training

### PR DESCRIPTION
**Pull request by operramon**

### CODE REVIEW

File: apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.html [hunk 1]
```diff
@@ -11,13 +11,24 @@ <h2>{{ 'stash.settings.processes.' + type | translate }}</h2>
...
+        </pa-button>
+        <div class="selection">{{ selectedLabelsets[type].join(', ') }}</div>
+      </div>
+
+      <pa-dropdown #dropdown
+                   (onOpen)="open[type] = true"
+                   (onClose)="open[type] = false">
+        <pa-option *ngFor="let labelset of labelsets | async"
+                   dontCloseOnSelect>
+        <pa-checkbox [value]="selectedLabelsets[type].includes(labelset.value)"
```
Comment (mpellerin42) [inline apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.html:28]: in the design you showed me, there was an option to select all the labelsets. Is this something we don't want at the end?

Comment (operramon) [reply apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.html]: oops, I forgot. Its added now.

File: apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.html [hunk 2]
```diff
@@ -23,10 +23,18 @@ <h2>{{ 'stash.settings.processes.' + type | translate }}</h2>
...
                    (onOpen)="open[type] = true"
                    (onClose)="open[type] = false">
+        <pa-option class="toggle-all"
+                   *ngIf="labelsets | async as labelsets"
+                   dontCloseOnSelect>
+          <pa-checkbox [value]="selectedLabelsets[type].length === labelsets.length"
+                       (change)="toggleAll(type)">
+            {{ 'stash.settings.processes.training.all_labelsets' | translate }}
+          </pa-checkbox>
+        </pa-option>
```
Comment (mpellerin42) [inline apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.html:33]: you can add a `<pa-separator></pa-separator>` below your option to add the line you see in the design:
image

File: apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.scss [hunk 3]
```diff
@@ -41,6 +41,10 @@ h2 {
   }
 }
 
+.toggle-all {
+  border-bottom: 1px solid $color-neutral-lighter;
+}
```
Comment (mpellerin42) [inline apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.scss:46]: with the pa-separator you won't need this

File: apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.ts [hunk 4]
```diff
@@ -52,21 +53,31 @@ export class KnowledgeBoxProcessesComponent implements OnInit, OnDestroy {
...
       this.selectedLabelsets[type] = [...this.selectedLabelsets[type], labelset];
     }
     this.cdr?.markForCheck();
   }
 
+  toggleAll(type: TrainingType) {
+    this.labelsets.pipe(take(1)).subscribe((labelsets) => {
+      const selectAll = this.selectedLabelsets[type].length < labelsets.length;
+      this.selectedLabelsets[type] = selectAll ? labelsets.map((item) => item.value) : [];
+      this.cdr?.markForCheck();
```
Comment (mpellerin42) [inline apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.ts:67]: FYI pastanaga provides a helper function doing some extra checks. You can use it by doing `markForCheck(this.cdr)`.

**Review by mpellerin42 — APPROVED** [review]

File: apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.html (hunk 2 above)
Comment (operramon) [reply apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.html:32]: Fixed 👍

File: apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.scss (hunk 3 above)
Comment (operramon) [reply apps/dashboard/src/app/knowledge-box/knowledge-box-process/knowledge-box-processes.component.scss]: Fixed

**Pull request merged**